src/openads/application/service/DisplayPositionUseCase.js

```diff
@@ -0,0 +1,30 @@
+import {POSITION_VISIBLE} from '../../domain/position/positionStatus'
+
+export default class DisplayPositionUseCase {
+  /**
+   * @constructor
+   * @param {PositionRepository} positionRepository
+   */
+  constructor ({positionRepository}) {
+    this._positionRepository = positionRepository
+  }
+
+  /**
+   * Displays a position in the page
+   * @param {string} position id
+   * @return {Promise<Position>}
+   */
+  displayPosition ({id} = {}) {
+    return this._positionRepository.find({id})
+      .then(this._resolveOptionalPosition)
```

Comment:
this line works?
` .then(this._resolveOptionalPosition)`

wouldn't it be ?
` .then(() => this._resolveOptionalPosition)`

Reply:
It's magical, right?

![](https://media.giphy.com/media/12NUbkX6p4xOO4/giphy.gif)

It seems that when there is only 1 parameter received and from a promise in a then clause you can use it directly to call a method which also receives just 1 parameter in this way. Something like clean lambdas in Java. Notice that this only works with non-named-parameters. 

You can try it here:
https://jsbin.com/lehilegupe/edit?js,console,output

It seems to be working fine as tests are covering this and are passing.

Reply:
The bin does not contain a related example :(
I tried that before commenting, and didn't work..

Reply:
Sorry, bin is missing. Anyway, in order to raise custom exceptions, I will be forced to modify this trick, so it will not be used... :(